src/main/kotlin/com/pubnub/internal/v2/subscription/SubscriptionSetImpl.kt

```diff
@@ -22,6 +28,43 @@ internal class SubscriptionSetImpl(
     private val pubnub: PubNub,
     initialSubscriptions: Set<SubscriptionImpl> = emptySet(),
 ) : SubscriptionSet(), EventEmitter {
+
+    override var onMessage: ((PubNub, PNMessageResult) -> Unit)? = null
```

Comment:
all these could be delegated properties:
```
    override var onMessage: ((PubNub, PNMessageResult) -> Unit)? by eventEmitter::onMessage
    override var onPresence: ((PubNub, PNPresenceEventResult) -> Unit)? by eventEmitter::onPresence
    override var onSignal: ((PubNub, PNSignalResult) -> Unit)? by eventEmitter::onSignal
    override var onMessageAction: ((PubNub, PNMessageActionResult) -> Unit)? by eventEmitter::onMessageAction
    override var onObjects: ((PubNub, PNObjectEventResult) -> Unit)? by eventEmitter::onObjects
    override var onFile: ((PubNub, PNFileEventResult) -> Unit)? by eventEmitter::onFile
```

Reply:
Nice, so concise. I will change it.